dev/bots/tool_subsharding.dart

```diff
@@ -49,7 +49,13 @@ class TestFileReporterResults {
     final List<String> errors = <String>[];
 
     for (final String metric in metrics.readAsLinesSync()) {
-      final Map<String, Object?> entry = json.decode(metric) as Map<String, Object?>;
+      /// Using print within a test adds the printed content to the json file report
+      /// as \u0000 making the file parsing step fail. The content of the json file
+      /// is expected to be a json dictionary per line and the following line removes
+      /// all the additional content at the beginning of the line until it finds the
+      /// first opening curly bracket.
+      final String sanitizedMetric = metric.replaceAll(RegExp(r'$.+{'), '{');
```

Comment:
I think this pattern should start with `^` for start of line, whereas `$` is end of line

Reply:
The stdout from print is added to the beginning of the line followed by the real event dict. e.g `@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@    ^@^@{"testID":11,"result":"success","skipped":false,"hidden":false,"type":"testDone","time":22248}`

Reply:
right but `^.*{` should catch that (changed a + to *, since we're not guaranteed to have a newline)

Reply:
Updated!